src/editor.c

```diff
@@ -1683,6 +1685,64 @@ static void close_block(GeanyEditor *editor, gint pos)
 /* checks whether @p c is an ASCII character (e.g. < 0x80) */
 #define IS_ASCII(c) (((unsigned char)(c)) < 0x80)
 
+/* Similar to read_current_word it will read the current scope of the word at the cursor position
+ * and writes it into the given buffer. The buffer will be NULL terminated in any case,
+ * even when the word is truncated because worlden is too small. psoition can be -1, then
+ * the current position is used.
+ * wc are the wordchars to use, if NULL, GEANY_WORDCHARS will be used */
+static void read_current_scope(GeanyEditor *editor, gint pos, gchar *word, gsize wordlen,
+		const gchar *wc)
+{
+	gint line, line_start, startword, endword;
+	gchar *chunk;
+	ScintillaObject *sci;
+
+	g_return_if_fail(editor != NULL);
+	sci = editor->sci;
+
+	if (pos == -1)
+		pos = sci_get_current_position(sci);
+
+	line = sci_get_line_from_position(sci, pos);
+	line_start = sci_get_position_from_line(sci, line);
+	startword = pos - line_start;
+	endword = pos - line_start;
+
+	word[0] = '\0';
+	chunk = sci_get_line(sci, line);
+
+	const char *punctuation = ":.";
```

Comment:
I haven't really looked over the code (existing or your changes), but couldn't it lookup the tag for the type of the left hand side, and use that as a the scope, rather than assuming the LHS _is_ the scope? Like if the LHS was a function call, it would use the return type or if it was an instance, it would use the type of its declaration.